src/System/Logging/Facade/Sink.hs

```diff
@@ -27,9 +28,12 @@ getLogSink = readIORef logSink
 setLogSink :: LogSink -> IO ()
 setLogSink = atomicWriteIORef logSink
 
--- | A log sink that writes log messages to `stderr`
-defaultLogSink :: LogSink
-defaultLogSink record = hPutStrLn stderr output
+-- | A thread-safe log sink that writes log messages to `stderr`
+defaultLogSink :: IO LogSink
+defaultLogSink = defaultLogSink_ `fmap` newMVar ()
```

Comment:
Technically, I agree that I want to use `<$>` here.  But I think if you want backward compatible `-Wall`-sane code, this would require either `CPP` or a dependency on `base-compat`.